flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/allocator/JobAllocationsInformation.java

```diff
@@ -0,0 +1,100 @@
+import org.apache.flink.runtime.executiongraph.ExecutionJobVertex;
+import org.apache.flink.runtime.executiongraph.ExecutionVertex;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.state.KeyGroupRange;
+import org.apache.flink.runtime.state.KeyGroupRangeAssignment;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+
+/** Information about allocations of Job Vertices. */
+@Internal
+public class JobAllocationsInformation {
+
+    private final Map<JobVertexID, List<VertexAllocationInformation>> vertexAllocations;
+
+    JobAllocationsInformation(
+            Map<JobVertexID, List<VertexAllocationInformation>> vertexAllocations) {
+        this.vertexAllocations = vertexAllocations;
+    }
+
+    public static JobAllocationsInformation fromGraph(@Nullable ExecutionGraph graph) {
+        return graph == null ? empty() : new JobAllocationsInformation(calculateAllocations(graph));
+    }
+
+    public List<VertexAllocationInformation> getAllocations(JobVertexID jobVertexID) {
+        return vertexAllocations.getOrDefault(jobVertexID, emptyList());
+    }
+
+    private static Map<JobVertexID, List<VertexAllocationInformation>> calculateAllocations(
+            ExecutionGraph graph) {
+        final Map<JobVertexID, List<VertexAllocationInformation>> allocations = new HashMap<>();
+        for (ExecutionJobVertex vertex : graph.getVerticesTopologically()) {
+            JobVertexID jobVertexId = vertex.getJobVertexId();
+            for (ExecutionVertex executionVertex : vertex.getTaskVertices()) {
+                AllocationID allocationId =
+                        executionVertex.getCurrentExecutionAttempt().getAssignedAllocationID();
+                KeyGroupRange kgr =
+                        KeyGroupRangeAssignment.computeKeyGroupRangeForOperatorIndex(
+                                vertex.getMaxParallelism(),
+                                vertex.getParallelism(),
+                                executionVertex.getParallelSubtaskIndex());
+                allocations
+                        .computeIfAbsent(jobVertexId, ignored -> new ArrayList<>())
+                        .add(new VertexAllocationInformation(allocationId, jobVertexId, kgr));
+            }
+        }
+        return allocations;
+    }
+
+    public static JobAllocationsInformation empty() {
+        return new JobAllocationsInformation(emptyMap());
+    }
+
+    public boolean isEmpty() {
+        return vertexAllocations.isEmpty();
+    }
+
+    /** Information about the allocations of a single Job Vertex. */
+    public static class VertexAllocationInformation {
+        public final AllocationID allocationID;
+        public final JobVertexID jobVertexID;
+        public final KeyGroupRange keyGroupRange;
```

Comment:
Should we have getters here instead? 🙄 

```suggestion
        public final AllocationID allocationID;
        public final JobVertexID jobVertexID;
        public final KeyGroupRange keyGroupRange;
```

Reply:
Done in 06dede2f641afcb9f4c1d5129806d52d3cbf08a4.